### PR TITLE
BaSP-TG-85: Employee redux - Create, update and delete

### PR DESCRIPTION
**Pull request by renzopimentel**
- Create constants to create, edit and update employees
- Create actions to create, update and delete a employee
- Create thunks to create, update and delete a employee
- Complete task reducer to create, update and delete a employee
- Connect list component with store to get, update, create and delete employee

For QAs:
See "how to test?" on Trello:
[TG-85](https://trello.com/c/6fSRmdkj)

### CODE REVIEW

File: src/redux/employees/reducer.js [hunk 1]
```diff
@@ -34,7 +47,90 @@ const employeesReducer = (state = INITIAL_STATE, action) => {
...
+        ...state,
+        isLoading: false,
+        error: action.payload,
+        modalContent: { title: 'ERROR!', content: `Could not UPDATE Employee! ${action.payload}` },
+        showModalMessage: true
+      };
+    case CONFIRM_MODAL_OPEN:
+      return {
+        ...state,
+        ...state,
```
Comment (CristianBucci) [inline src/redux/employees/reducer.js:116]: Delete this duplicate line!

File: src/Components/Employees/Form/index.js [hunk 2]
```diff
@@ -1,137 +1,95 @@
...
-          content: `Could not update Employee! ${response.message}`
-        });
-        setShowModalMessage(true);
-      }
-    } catch (error) {
-      setModalContent({ title: 'ERROR!', content: `Could not update Employee! ${error.message}` });
-      setShowModalMessage(true);
-    }
+  const modalFunction = () => {
+    modalContent.title.includes('SUCCESS') ? redirect() : null;
```
Comment (MaxiG-dev) [inline src/Components/Employees/Form/index.js:75]: You can replace 
`modalContent.title.includes('SUCCESS') ? redirect() : null;`
with
`modalContent.title.includes('SUCCESS') && redirect();`

File: src/Components/Employees/index.js [hunk 3]
```diff
@@ -1,57 +1,72 @@
...
-  let delParams = {
-    id: itemId,
-    path: 'employees',
-    list: employeesList
+  const redirect = () => {
+    props.history.push('/employees');
+  };
+
+  const modalFunction = () => {
+    modalContent.title.includes('SUCCESS') ? redirect() : null;
```
Comment (MaxiG-dev) [inline src/Components/Employees/index.js:52]: You can replace 
`modalContent.title.includes('SUCCESS') ? redirect() : null;`
with
`modalContent.title.includes('SUCCESS') && redirect();`

Comment (renzopimentel) [reply src/Components/Employees/index.js:52]: Perfect, done!

File: src/Components/Employees/Form/index.js (hunk 2 above)
Comment (renzopimentel) [reply src/Components/Employees/Form/index.js]: Done!

**Review by MaxiG-dev — APPROVED** [review]
Looks great for me!

**Review by melissabonillasellanes — APPROVED** [review]
Everything looks fine!

**Review by matescarabino — APPROVED** [review]
LGTM

**Review by nicolasblasco — APPROVED** [review]

**Review by julianpiaggio — APPROVED** [review]
🕵️‍♂️ QA FEEDBACK

First of all, we wanted to tell you that we consider that your pr complies
the requirements to be approved, and that it can be merged into the master branch 👍

Now, if we consider leaving you some comments:

* In the create, the message of the modal that responds when invalid data is entered is very generic, it does not specify what the error is or in what field. This can generate confusion, and the ideal would be that when confirming the action and that the modal responds, it can provide us with information about what or where we are putting incorrect data.

 
BaSP-TG-85-Employee redux-modal

* when the delete is confirmed, we don't have a modal with a success message. Ideally, a response should appear with the text "employee successfully deleted"

Here we leave you a detailed link with a video that describes what we tell you:
https://imgur.com/a/QaHTABu

**Pull request merged**